hazelcast/src/main/java/com/hazelcast/transaction/impl/TransactionLog.java

```diff
@@ -155,7 +156,7 @@ private void invokeAsync(NodeEngine nodeEngine, BiConsumer callback,
             operationService.invokeOnTarget(op.getServiceName(), op, target);
         } else {
             operationService.invokeOnPartitionAsync(op.getServiceName(), op, op.getPartitionId())
-                    .whenCompleteAsync(callback);
+                    .whenCompleteAsync(callback, nodeEngine.getExecutionService().getExecutor(ExecutionService.ASYNC_EXECUTOR));
```

Comment:
Shouldn't we continue to use FJP common here? Reasoning is: If execution is triggered by user we always should go with FJP common, otherwise we should go with internal async executor.

Reply:
Make sense, thanks!